### PR TITLE
DENG-2918-UDFs for URL parsing

### PR DESCRIPTION
**Pull request by lelilia**
Adding new UDFs that I need going forward. 

Parsing the host url and the path from a url string

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-2920)

### CODE REVIEW

File: sql/mozfun/utils/get_url_path/udf.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+CREATE OR REPLACE FUNCTION utils.get_url_path(url STRING)
+RETURNS STRING AS (
+  "/" || COALESCE(
+    REGEXP_EXTRACT(REPLACE(REPLACE(url, "https://", ""), "http://", ""), r"\/([^&?#]*)"),
```
Comment (kik-kik) [inline sql/mozfun/utils/get_url_path/udf.sql:4]: Can we foresee any other protocols than HTTP and HTTPS making it through for example `ftp://`?

**Review by kwindau — APPROVED** [review]
LGTM

File: sql/mozfun/utils/get_url_host/udf.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+CREATE OR REPLACE FUNCTION utils.get_url_host(url STRING)
```
Comment (edugfilho) [inline sql/mozfun/utils/get_url_host/udf.sql:1]: I believe [NET.HOST(url)](https://cloud.google.com/bigquery/docs/reference/standard-sql/net_functions#nethost) can achieve this already :)

Comment (lelilia) [reply sql/mozfun/utils/get_url_host/udf.sql:1]: woooooow thank you that is way easier \o/

File: sql/mozfun/utils/get_url_host/udf.sql [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    SPLIT(REPLACE(REPLACE(REPLACE(url, "android-app://", ""), "http://", ""), "https://", ""), "?")[
+      SAFE_OFFSET(0)
+    ],
+    "/"
+  )[SAFE_OFFSET(0)]
+);
+
+-- Tests
+SELECT
+  mozfun.assert.equals("some-url.com", utils.get_url_host("https://some-url.com/path?something")),
```
[the remark this answers is not in the document]
Comment (lelilia) [reply sql/mozfun/utils/get_url_host/udf.sql:13]: irrelevant after I learned about net.host() from eduardo \o/

File: sql/mozfun/utils/get_url_path/udf.sql (hunk 1 above)
Comment (lelilia) [reply sql/mozfun/utils/get_url_path/udf.sql:4]: not in my use case ... future people can improve upon it ;)

**Review by edugfilho — APPROVED** [review]

**Pull request merged**